src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandParserSupport.cs

```diff
@@ -0,0 +1,222 @@
+                                            Accept.ExactlyOneArgument());
+                    }
+
+                    paramOptionList.Add(option);    // add the option
+                    templateParamMap.Add(canonical, assignedAliases.ToList());   // map the template canonical name to its aliases.
+                    takenAliases.UnionWith(assignedAliases);  // add the aliases to the taken aliases
+                }
+                else
+                {
+                    invalidParams.Add(canonical);
+                }
+            }
+
+            if (invalidParams.Count > 0)
+            {
+                string unusableDisplayList = string.Join(", ", invalidParams);
+                throw new Exception($"Template is malformed. The following parameter names are invalid: {unusableDisplayList}");
+            }
+
+            return GetNewCommandForTemplate(commandName, templateName, NewCommandVisibleArgs, NewCommandHiddenArgs, DebuggingCommandArgs, paramOptionList.ToArray());
+        }
+
+        private static Option[] NewCommandVisibleArgs
+        {
+            get
+            {
+                return new[]
+                {
+                    Create.Option("-l|--list", LocalizableStrings.ListsTemplates, Accept.NoArguments()),
+                    Create.Option("-lang|--language", LocalizableStrings.LanguageParameter,
+                                    // TODO: dynamically get the language set for all installed templates.
+                                    Accept.ExactlyOneArgument()
+                                        .WithSuggestionsFrom("C#", "F#")
+                                        .With(defaultValue: () => "C#")),
+                    Create.Option("-n|--name", LocalizableStrings.NameOfOutput, Accept.ExactlyOneArgument()),
+                    Create.Option("-o|--output", LocalizableStrings.OutputPath, Accept.ExactlyOneArgument()),
+                    Create.Option("-h|--help", LocalizableStrings.DisplaysHelp, Accept.NoArguments()),
+                    Create.Option("--type", LocalizableStrings.ShowsFilteredTemplates, Accept.ExactlyOneArgument()),
+                    Create.Option("--force", LocalizableStrings.ForcesTemplateCreation, Accept.NoArguments()),
+                };
+            }
+        }
+
+        private static Option[] NewCommandHiddenArgs
+        {
+            get
+            {
+                return new[]
+                {
+                    Create.Option("-a|--alias", string.Empty, Accept.ExactlyOneArgument()),
+                    Create.Option("-x|--extra-args", string.Empty, Accept.OneOrMoreArguments()),
+                    Create.Option("--locale", string.Empty, Accept.ExactlyOneArgument()),
+                    Create.Option("--quiet", string.Empty, Accept.NoArguments()),
+                    Create.Option("-i|--install", string.Empty, Accept.OneOrMoreArguments()),
```

Comment:
Having the separate Visible & Hidden collections is mainly to keep track of what gets included in the command help output. Having the help text (2nd param) as string.Empty would also keep the hidden ones from showing up. 

The reserved list is only used for preventing those reserved flag values from getting assigned with the template param flags get assigned.